zendesk/trigger.go

```diff
@@ -40,7 +40,7 @@ type Trigger struct {
 
 // TriggerAPI an interface containing all trigger related methods
 type TriggerAPI interface {
-	GetTriggers() ([]Trigger, Page, error)
+	GetTriggers(page int) ([]Trigger, Page, error)
```

Comment:
The change looks fine but i worry that it might be a tad short sighted.

there are alot of options that can potentially be passed to this api. https://developer.zendesk.com/rest_api/docs/support/triggers#list-triggers

Are we worried at all about the where this pattern might ulitmately take us if we were to want to pass in some of the other api options i.e: are we ok with this method taking 4 or 5 parameters to call?

 Here is a client that i have used before that i really like. https://github.com/guregu/dynamo

It makes all of the api functions return builders that can be used to set various options in a backward compatible way.

Just curious what your thoughts are. If you don't think it is necessary this change looks fine.

Reply:
I think method chain is better for usability. But, it would make `zendesk.Client`'s state much bigger because our library has all API methods in one instance.

What do you think about taking parameter as struct pointer?
```go
type TriggersParameter struct {
  PageNum int
  // other parameters
}

type TriggerAPI interface {
	GetTriggers() ([]Trigger, Page, error)
	GetTriggers(params *TriggersParameters) ([]Trigger, Page, error)
```

Then, we could simply pass `nil` as the argument if parameters are unnecessary, like this.
```go
client.GetTriggers(nil)
```

Reply:
could we just pass the zero value in if we didn't care about the parameters?

Reply:
I decided to take parameter struct approach, because I think it more testable than method chain.

And I found a reference case in go-github library.
https://github.com/google/go-github/blob/master/github/issues.go#L91

> could we just pass the zero value of the struct in if we didn't care about the parameters?

In this case, we should not use `TriggersParameter{}` to mean **no parameter** because the `PageNum` is initialized by `0`, and it would end up as `?page_num=0`.

So, I think pointer nil check is better than validation of all fields in TriggerParameter.